### PR TITLE
Move .travis.yml over to travis-complex.yml default with some changes

### LINKED ISSUE #5: Set up CI and automatic releases
Basics
- [x] Build matrix with GHCs 8.0.1 on up
- [ ] Build matrix with `mafia build`
- [x] Build matrix with `cabal build`
- [x] Build matrix with `stack build`
- [x] Build matrix with OS X
- [x] Build matrix with Windows

Lazymode
- [ ] Publish haddocks to gh-pages on master
- [ ] Publish tagged versions to Hackage
- [x] Publish Linux binaries to gh-releases (on tag)
- [x] Publish OS X binaries to gh-releases (on tag)
- [ ] Publish Windows binaries to gh-releases (on tag)

### PR DESCRIPTION
**Pull request by justanotherdot**
This moves over to a prebuilt .travis.yaml with a few minor changes and
hopes to resolve #5. It also keeps stack.yaml at LTS 9.2 for the time
being and includes builds for OS X but not Windows. This does not
include:

1. Publishing haddock docs to github-pages
2. Publishing tagged versions of hackage
3. Publishing linux binaries to github-releases (on tag)
4. Publishing OS X binaries to github-releases (on tag)
5. Publishing Windows binaries to github-release (on tag)

The alternative to using the stack.yaml provided is to do 'stack init &&
stack build' but that may lead to breaking changes with certain libs
(exact-print etl. al); see #9.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,38 +1,124 @@
...
   include:
-    - compiler: "ghc-8.0.1"
-      addons: {apt: {packages: [ghc-ppa-tools,cabal-install-1.24,ghc-8.0.1], sources: [hvr-ghc]}}
-    - compiler: "ghc-8.0.2"
-      addons: {apt: {packages: [ghc-ppa-tools,cabal-install-1.24,ghc-8.0.2], sources: [hvr-ghc]}}
-    - compiler: "ghc-8.2.1"
-      addons: {apt: {packages: [ghc-ppa-tools,cabal-install-1.24,ghc-8.0.2], sources: [hvr-ghc]}}
+  # The Cabal builds.
+  - env: BUILD=cabal GHCVER=8.0.2 CABALVER=1.24 HAPPYVER=1.19.5 ALEXVER=3.1.7
+    compiler: ": #GHC 8.0.2"
```
Comment (thumphries) [inline .travis.yml:15]: Could we also get a 8.2.1 build?

**Comment by thumphries** [comment]
Thanks so much for doing this, looks legit!

**Comment by justanotherdot** [comment]
Crossing my fingers travis is happy with it!

Maybe we should ticket 1 and 2+3+4 as separate tickets? Forstall 5 until actual windows support comes into play. What's your thoughts?

**Comment by vaibhavsagar** [comment]
I think AppVeyor is the way to go for Windows builds, and that should be a separate PR.

**Comment by thumphries** [comment]
> Maybe we should ticket 1 and 2+3+4 as separate tickets? Forstall 5 until actual windows support comes into play. What's your thoughts?

If you mean the more subtle / nice-to-have points in #5, I'm just going to mark stuff off the checklist. It's not quite as cathartic as a closed Github issue, I know, but issue fatigue is a thing.

File: .travis.yml (hunk 1 above)
Comment (justanotherdot) [reply .travis.yml:15]: Sure thing, I'll just need to work out how

Comment (justanotherdot) [reply .travis.yml:15]: Should we just do 8.2.1 instead of 8.0.2? Or both in tandem?

Comment (thumphries) [reply .travis.yml:15]: Looks like just copying this entry and replacing the strings `8.0.2` -> `8.2.1` will work!

Comment (justanotherdot) [reply .travis.yml:15]: Nice! How did you test that so quickly?

Comment (vaibhavsagar) [reply .travis.yml:15]: Would we need Cabal 2 for `8.2.1` or is `1.24` fine?

Comment (thumphries) [reply .travis.yml:15]: I'm currently using 1.24 with 8.2.1 without any problems.

Comment (justanotherdot) [reply .travis.yml:15]: Looks like it built fine; just waiting on `stack default` for OS X  now.

**Comment by thumphries** [comment]
Thanks Ryan, I really appreciate your work on this!

**Comment by justanotherdot** [comment]
No worries! Onto another issue ...

**Pull request merged**